File: minigit-4/py/commands/ls_files.py
"""ls-files command - List indexed files."""

import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).parent.parent))

from utils import find_repo_root
from index import Index


def run(args):
    """List indexed files."""
    repo_root = find_repo_root()
    if repo_root is None:
        print("fatal: not a minigit repository", file=sys.stderr)
        return 1

    # Parse arguments
    staged = False

    for arg in args:
        if arg in ('--stage', '--staged', '-s'):
            staged = True

    # Read index
    index = Index.read(repo_root)

    for entry in index.get_sorted_entries():
        if staged:
            mode_str = f'{entry.mode:o}'
            # Stage is always 0 for normal entries
            print(f'{mode_str} {entry.sha} 0\t{entry.name}')
        else:
            print(entry.name)

    return 0

File: minigit-4/ts/src/commands/ls-files.ts
// ls-files command - List indexed files

import { findRepoRoot } from '../utils';
import { readIndex } from '../index-file';

export function lsFiles(args: string[]): number {
  const repoRoot = findRepoRoot();
  if (!repoRoot) {
    console.error('fatal: not a minigit repository');
    return 1;
  }

  const showStaged = args.includes('--staged') || args.includes('-s');

  const entries = readIndex(repoRoot);

  for (const entry of entries) {
    if (showStaged) {
      // Format: mode sha stage path
      const modeStr = entry.mode.toString(8).padStart(6, '0');
      const stage = (entry.flags >> 12) & 0x3;
      console.log(`${modeStr} ${entry.sha} ${stage}\t${entry.name}`);
    } else {
      console.log(entry.name);
    }
  }

  return 0;
}
